lib/spack/spack/cmd/find.py

```diff
@@ -114,6 +123,13 @@ def query_arguments(args):
     if args.implicit:
         explicit = False
     q_args = {'installed': installed, 'known': known, "explicit": explicit}
+
+    # Time window of installation
+    for attribute in ('start_date', 'end_date'):
+        date = getattr(args, attribute)
+        if date:
+            q_args[attribute] = spack.database.str2datetime(date)
```

Comment:
Probably should add a catch if data is ill formatted that should raise an Exception telling which date was ill formed and how it should have been formatted

Reply:
Uh? Is there something wrong with the current behavior?
```console
$ spack find --start-date=456
==> Error: time data '456' does not match format '%Y-%m-%d %H:%M:%S'
```

Reply:
I guess I'm just still burned from the `SyntaxError in package.py`... At the moment it is only used on the command line, where the user should immediately see what happened, I worry about a future use in less user-obvious situations

Reply:
Can you elaborate on the comment above? I don't get the request there...

Reply:
When you have a syntax error (I'm not too sure anymore of which kind it has to be) in either one of the configuration files or any `package.py` in the repo, there were at least cases where spack wouldn't react to any `spack install` or `spack spec` like command except printing out something to the effect of `SyntaxError in 'package.py'`, I'd have wanted there to be a feedback in the sense of `SyntaxError in 'path-to-package.py'`.
In the same spirit: I'd like to get feedback on which _argument_ is broken. Not how the broken context looks like. If it's only used from the command line it probably won't matter much but I spend more time with finding out which `package.py` was broken than was good for my sanity :D